crates/acvm/src/lib.rs

```diff
@@ -282,11 +284,35 @@ pub trait ProofSystemCompiler {
 
 /// Supported NP complete languages
 /// This might need to be in ACIR instead
+#[derive(Debug, Clone)]
 pub enum Language {
     R1CS,
     PLONKCSat { width: usize },
 }
 
+pub trait CustomGate {
+    fn support(&self, opcode: &str) -> bool;
+    fn support_gate(&self, gate: &Gate) -> bool;
+}
+
+impl CustomGate for Language {
+    fn support(&self, _opcode: &str) -> bool {
```

Comment:
`supports` would read better in `self.supports(...)`

Reply:
Done